barretenberg/cpp/src/barretenberg/vm/generated/spike_verifier.cpp

```diff
@@ -0,0 +1,110 @@
+
+
+#include "./spike_verifier.hpp"
+#include "barretenberg/commitment_schemes/zeromorph/zeromorph.hpp"
+#include "barretenberg/numeric/bitop/get_msb.hpp"
+#include "barretenberg/polynomials/polynomial.hpp"
+#include "barretenberg/transcript/transcript.hpp"
+
+namespace bb {
+
+SpikeVerifier::SpikeVerifier(std::shared_ptr<Flavor::VerificationKey> verifier_key)
+    : key(verifier_key)
+{}
+
+SpikeVerifier::SpikeVerifier(SpikeVerifier&& other) noexcept
+    : key(std::move(other.key))
+    , pcs_verification_key(std::move(other.pcs_verification_key))
+{}
+
+SpikeVerifier& SpikeVerifier::operator=(SpikeVerifier&& other) noexcept
+{
+    key = other.key;
+    pcs_verification_key = (std::move(other.pcs_verification_key));
+    commitments.clear();
+    return *this;
+}
+
+using FF = SpikeFlavor::FF;
+
+// Evaluate the given public input column over the multivariate challenge points
+[[maybe_unused]] FF evaluate_public_input_column(std::vector<FF> points, std::vector<FF> challenges)
+{
+    Polynomial<FF> polynomial(points);
+    return polynomial.evaluate_mle(challenges);
+}
+
+/**
+ * @brief This function verifies an Spike Honk proof for given program settings.
+ *
+ */
+bool SpikeVerifier::verify_proof(const HonkProof& proof, const std::vector<FF>& public_inputs)
+{
+    using Flavor = SpikeFlavor;
+    using FF = Flavor::FF;
+    using Commitment = Flavor::Commitment;
+    // using PCS = Flavor::PCS;
+    // using ZeroMorph = ZeroMorphVerifier_<PCS>;
+    using VerifierCommitments = Flavor::VerifierCommitments;
+    using CommitmentLabels = Flavor::CommitmentLabels;
+
+    RelationParameters<FF> relation_parameters;
+
+    transcript = std::make_shared<Transcript>(proof);
+
+    VerifierCommitments commitments{ key };
+    CommitmentLabels commitment_labels;
+
+    const auto circuit_size = transcript->template receive_from_prover<uint32_t>("circuit_size");
+
+    if (circuit_size != key->circuit_size) {
+        return false;
+    }
+
+    // Get commitments to VM wires
+    commitments.Spike_kernel_inputs =
```

Comment:
Probably it does not hurt, but I thought that the prover would not generate the commitment for public columns.

Reply:
You are correct, the pr description goes over the current approach, as a temporary step this is enough. Otherwise we would need to hash each public input individually.

Reply:
@Maddiaa0  Ok. Could you just make sure we keep a ticket or a reminder somewhere about this?